### PR TITLE
Add "Edit and Save" doc explaining how each work and their props.

### PR DESCRIPTION
**Pull request by mtias**
image

Initial document for Edit and Save.

### CODE REVIEW

File: docs/block-edit-save.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+It operates the same way as it does on `edit` and allows to save certain attributes directly to the markup, so they don't have to be computed on the server. This is how most static blocks are expected to work.
+
+```js
+// Defining the edit interface
+save( { attributes } ) {
+	return <div>{ attributes.content }</div>;
+}
+```
+
+Attributes that are not used in `save` are encoded within the comment delimiters used to recognize blocks.
```
Comment (youknowriad) [inline docs/block-edit-save.md]: not totally true, this depends on the `attributes` definition instead.

Comment (mtias) [reply docs/block-edit-save.md]: True, I was just on old gears.

File: docs/block-edit-save.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+The `edit` function describes the structure of your block in the context of the editor. This represents what the editor will render when the block is used.
+
+```js
+// Defining the edit interface
+edit() {
+	return <hr />;
+}
+```
+
+This function can return properties through an object argument.
```
Comment (aduth) [inline docs/block-edit-save.md]: I'm not sure what's meant by this sentence. Is this trying to describe the behavior of props? In which case, "this function _receives_ properties" is more accurate than "can return".

File: docs/block-edit-save.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+### attributes
+
+This property returns all the available attributes and their values, as described by the `attributes` property when the block type was registered. In this case, assuming we had defined an attribute of `content` during block registration, we would receive and use that value in our edit function:
```
Comment (aduth) [inline docs/block-edit-save.md]: Another odd use of "returns". I might expect "makes available", "exposes", "lists", "contains".

File: docs/block-edit-save.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+			{ attributes.content }
+			{ focus &&
+				<span>Shows only when the block is focused.</span>
+			}
+		</div>
+	);
+}
+```
+
+Expand on how to focus specific elements within the block...
```
Comment (aduth) [inline docs/block-edit-save.md]: Do we want to leave this here? Maybe make more clear that it's a _TODO_ if we're leaving it for another pull request?

File: docs/block-edit-save.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+### setAttribute
+
+This function allows you to update individual attributes for the block based on user interactions.
+
+```js
+// Defining the edit interface
+edit( { attributes, className, focus } ) {
+	// Simplify accessing attributes
+	{ content, mySetting } = attributes;
```
Comment (aduth) [inline docs/block-edit-save.md]: Needs `const` in assignment.

File: docs/block-edit-save.md [hunk 6]
```diff
@@ -0,0 +1,107 @@
+# Edit and Save
+
+When registering a block, the `edit` and `save` functions provide the interface for how a block is going to be rendered within the editor, how it will operate and be manipulated, and how it will be saved.
+
+## Edit
+
+The `edit` function describes the structure of your block in the context of the editor. This represents what the editor will render when the block is used.
+
+```js
```
Comment (aduth) [inline docs/block-edit-save.md]: Will you be adding ES5 variations?

File: docs/block-edit-save.md (hunk 2 above)
Comment (aduth) [reply docs/block-edit-save.md]: Do we want to shy away from the "props" terminology?

File: docs/block-edit-save.md (hunk 6 above)
Comment (mtias) [reply docs/block-edit-save.md]: Maybe.

Comment (nb) [reply docs/block-edit-save.md]: Given the average JS familiarity level and the uncertainty/complexity of build tools for plugins, I think we should prioritize adding interactive ES5/modern ES switcher for alternate versions of the examples.

File: docs/block-edit-save.md [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+// Defining the edit interface
+edit() {
```
Comment (nb) [inline docs/block-edit-save.md]: The code sample could use some context – maybe add a minimal block?

**Pull request merged**